pkg/controller/network/vpc/controller.go

```diff
@@ -97,131 +116,127 @@ type external struct {
+	// update the CRD spec for any new values from provider
+	current := cr.Spec.ForProvider.DeepCopy()
+	ec2.LateInitializeVPC(&cr.Spec.ForProvider, &observed)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
+	cr.Status.AtProvider = ec2.GenerateVpcObservation(observed)
 
 	return managed.ExternalObservation{
-		ResourceExists:    true,
-		ConnectionDetails: managed.ConnectionDetails{},
-		ResourceUpToDate:  ec2.IsUpToDate(cr.Spec.VPCParameters, observed),
+		ResourceExists:   true,
+		ResourceUpToDate: ec2.IsVpcUpToDate(cr.Spec.ForProvider, observed),
 	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
-	cr, ok := mgd.(*v1alpha3.VPC)
+	cr, ok := mgd.(*v1beta1.VPC)
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	// if VPCID already exists, skip creating the vpc
-	// this happens when an error has occurred when modifying vpc attributes
-	if meta.GetExternalName(cr) == "" {
-
-		req := e.client.CreateVpcRequest(&awsec2.CreateVpcInput{
-			CidrBlock: aws.String(cr.Spec.CIDRBlock),
-		})
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
 
-		rsp, err := req.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
-		}
-
-		meta.SetExternalName(cr, aws.StringValue(rsp.Vpc.VpcId))
-		if err := e.kube.Update(ctx, cr); err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errPersistExternalName)
-		}
+	result, err := e.client.CreateVpcRequest(&awsec2.CreateVpcInput{
+		CidrBlock:       aws.String(cr.Spec.ForProvider.CIDRBlock),
+		InstanceTenancy: awsec2.Tenancy(aws.StringValue(cr.Spec.ForProvider.InstanceTenancy)),
+	}).Send(ctx)
 
-		cr.SetConditions(runtimev1alpha1.Creating())
-		cr.UpdateExternalStatus(*rsp.Vpc)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
 
+	meta.SetExternalName(cr, aws.StringValue(result.Vpc.VpcId))
+
 	// modify vpc attributes
 	for _, input := range []*awsec2.ModifyVpcAttributeInput{
 		{
 			VpcId:            aws.String(meta.GetExternalName(cr)),
-			EnableDnsSupport: &awsec2.AttributeBooleanValue{Value: aws.Bool(cr.Spec.EnableDNSSupport)},
+			EnableDnsSupport: &awsec2.AttributeBooleanValue{Value: cr.Spec.ForProvider.EnableDNSSupport},
 		},
 		{
 			VpcId:              aws.String(meta.GetExternalName(cr)),
-			EnableDnsHostnames: &awsec2.AttributeBooleanValue{Value: aws.Bool(cr.Spec.EnableDNSHostNames)},
+			EnableDnsHostnames: &awsec2.AttributeBooleanValue{Value: cr.Spec.ForProvider.EnableDNSHostNames},
 		},
 	} {
-		attrReq := e.client.ModifyVpcAttributeRequest(input)
-
-		if _, err := attrReq.Send(ctx); err != nil {
+		if _, err := e.client.ModifyVpcAttributeRequest(input).Send(ctx); err != nil {
 			return managed.ExternalCreation{}, errors.Wrap(err, errModifyVPCAttributes)
 		}
 	}
```

Comment:
It seems like this PR mostly has a pattern of creating a resource in `Create`, then waiting until `Update` time to set any fields that are modifiable. Should we follow that pattern for VPCs?